.github/workflows/clang-tools-amd64.yml

```diff
@@ -74,7 +74,7 @@ jobs:
             # Uses a macOS arm image here
             # https://github.com/actions/runner-images
             runner: macos-14
-            os-cmake-args: '-DCMAKE_CXX_FLAGS="-static-libgcc -static-libstdc++ -flto" -DCMAKE_OSX_DEPLOYMENT_TARGET=10.15 ${POSIX_CMAKE_ARGS} ${MACOS_CMAKE_ARGS} -DLLVM_TARGETS_TO_BUILD=AArch64'
+            os-cmake-args: '-DCMAKE_CXX_FLAGS="-static-libgcc -static-libstdc++ -flto -DZSTD_STATIC_LINKING_ONLY" ${POSIX_CMAKE_ARGS} ${MACOS_CMAKE_ARGS} -DLLVM_TARGETS_TO_BUILD=AArch64'
```

Comment:
the error you are showing is about `libz.dylib` -- I don't see this flag documented anywhere other than in facebook/folly where it refers to the zstd compression library they use.

Assuming the intent is to link libz aka zlib statically you will probably have to supply both compiler and linker directives to accomplish this.

However I think you should investigate the root cause of why it cannot locate the libz.dylib which is meant to be supplied by macOS. My theory posted on the other thread is that there is a CPU architecture mismatch of some kind